src/anomalib/models/reverse_distillation/loss.py

```diff
@@ -23,8 +27,13 @@ def forward(self, encoder_features: list[Tensor], decoder_features: list[Tensor]
             Tensor: Cosine similarity loss
         """
         cos_loss = torch.nn.CosineSimilarity()
-        losses = list(map(cos_loss, encoder_features, decoder_features))
         loss_sum = 0
-        for loss in losses:
-            loss_sum += torch.mean(1 - loss)  # mean of cosine distance
+        for item in range(len(encoder_features)):
+            loss_sum += torch.mean(
+                1
+                - cos_loss(
+                    encoder_features[item].view(encoder_features[item].shape[0], -1),
+                    decoder_features[item].view(decoder_features[item].shape[0], -1),
+                )
+            )
```

Comment:
Based on the codacy comment above, maybe we could have something like this?
```suggestion
        for _idx, (encoder_feature, decoder_feature) in enumerate(zip(encoder_features, decoder_features)):
            loss_sum += torch.mean(
                1
                - cos_loss(
                    encoder_feature.view(encoder_feature.shape[0], -1),
                    decoder_feature.view(decoder_feature.shape[0], -1),
                )
            )
```